src/libraries/System.Data.Common/src/System/Xml/XPathNodePointer.cs

```diff
@@ -339,7 +337,7 @@ internal string Value
                 }
                 else if (_column.ColumnMapping == MappingType.Attribute || _fOnValue)
                 {
-                    DataRow row = Row;
+                    DataRow row = Row!;
```

Comment:
This is a possible NRE. `Row` currently could return a `null!` and the next line will fail.

Reply:
The way I understand the code is that when `_column` is non-null, `Row` will have a value (see the definition of `Row`). Sadly I don't know how to express this invariant. Marking `Row` as non-null seems worse though.